### DISCUSSION TITLE
Solidity 102-第20讲发送ETH中的测试题7为什么value是2 ETH，最终两个合约余额还能有1ETH

### DISCUSSION THREAD

**Discussion by HappyhappyChan** [AmazingAng/WTF-Solidity#762] [Q&A] [answered]
7. 假设存在如下两个合约(sendETH和ReceiveETH)，两个合约目前ETH余额皆为0，现在vitalik想通过SendETH合约的callETH函数往ReceiveETH合约转入1ETH，他将交易的value设置为2ETH，同时交易成功执行，那么此时sendETH合约和ReceiveETH的ETH余额分别为？答案是1ETH, 1ETH.
为什么呢？都设置了value是2ETH，为什么还能变成1ETH。不理解，求解答，感谢。
![image](https://github.com/user-attachments/assets/9a2c082c-b52b-4508-822b-0a804503cc8c)

**Comment by lxhyl** [comment]
题目的意思可能是`callETH()`的参数amount填的1

**Comment by DennisWang917** [comment, marked as answer]
这里的意思是部署完sendETH和receiveETH合约后，在左上方的Value那里输入2ETH
<img width="285" alt="Screenshot 2024-08-15 at 14 52 45" src="https://github.com/user-attachments/assets/45f6a855-045a-4a50-ae28-f1f0c6a0ef9c">
然后在sendETH合约中的callETH中输入1000000000000000000(因为这里的单位是wei，所以这里的1000000000000000000wei= 1ETH)
<img width="292" alt="Screenshot 2024-08-15 at 14 58 44" src="https://github.com/user-attachments/assets/e9b99cca-cfdf-47cb-99c6-491f57ec3757">
最后点击transact，就会得到sendETH合约receiveETH合约分别拥有1个ETH
<img width="278" alt="Screenshot 2024-08-15 at 15 00 16" src="https://github.com/user-attachments/assets/dda48ad2-fd9b-4a85-999b-0e28bc4b140c">
:)
